### PR TITLE
Migrated findResource method

### PR DESCRIPTION
**Pull request by skellamp**

### CODE REVIEW

File: module/VuFind/src/VuFind/Db/Service/CommentsService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/CommentsService.php:68]: I think you are almost certainly correct to make this false here and in other getters, but I'm curious what led you to make this change. Was it to address a specific problem, or just an improvement you decided to make along the way? (I'm just curious in case it was revealed by some other issue that it might be good for me to know about, or if you're just thinking ahead to optimization). Either way, thanks! :-)

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/ResourceService.php:106]: Now that migration of this method is complete, can we delete the equivalent method from the Laminas\Db code?

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/ResourceService.php:134]: This should say "resource" rather than "comment".

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/ResourceService.php:126]: I think this if statement can be simplified to: `$driver ??= $this->recordLoader->load($id, $source);`

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/ResourceService.php:212]: It looks like assignMetadata is only used in one other place, so it might be a good next step after this PR is merged to finish migrating the related logic in the UpgradeController so we can remove more code from the Laminas Resource classes.

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/ResourceService.php:218]: This is an existing problem, but we might as well fix it: it seems strange that we only do the substring on the primary title. It might be cleaner to:

```
$sortTitle = $driver->tryMethod('getSortTitle');
$title = mb_substr(
    !empty($sortTitle) ? $sortTitle : $driver->getBreadcrumb(),
    0,
    255,
    'UTF-8'
);
```

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/ResourceService.php:230]: We should make this `$driver->tryMethod('getPrimaryAuthor') ?? ''` to ensure that we always pass a string to `mb_substr`.

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (demiankatz) [inline module/VuFind/src/VuFind/Db/Service/ResourceService.php:241]: This could be simplified to `if (strlen($dates[0] ?? '') > 4) {`

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @skellamp, this is looking good. I haven't done hands-on testing yet, but I did a readthrough of the code and have a few small suggestions (many of which relate to possible improvements to the existing code, rather than changes that you made to the logic... but we might as well optimize while we're editing anyway).

Once you've had a chance to address these initial suggestions, I'll do some more interactive testing to see if I can find further issues.

Also, at this point, if you want to open multiple pull requests focused on different features, please feel free. I think it should be possible to work in different areas of the code so that we can move forward on multiple fronts without ending up with conflicts. Obviously, you don't have to make this more complicated/confusing than it needs to be, but if you're ever waiting for me to review something, feel free to start on something else, and I'll catch up as soon as I can!

File: module/VuFind/src/VuFind/Db/Service/CommentsService.php
Comment (skellamp) [reply module/VuFind/src/VuFind/Db/Service/CommentsService.php:68]: When the method was partially migrated, it did not have the functionality of creating a new record if it doesn't exist, so $create parameter that defaults to True was not passed. The logic in dev branch, is migrated as is, so I had pass the parameter to override the default action of creating a record if doesn't exist.

File: module/VuFind/src/VuFind/Db/Service/ResourceService.php
Comment (skellamp) [reply module/VuFind/src/VuFind/Db/Service/ResourceService.php:106]: There are some many other instances where this method is used that have other dependencies, at this moment it's better to keep this method.

Comment (demiankatz) [reply module/VuFind/src/VuFind/Db/Service/ResourceService.php:106]: Okay, that makes sense -- perhaps following this thread will reveal some other areas of refactoring that would be good next steps after this work is completed.

**Review by demiankatz — APPROVED** [review]
This all seems to be working, even after merging the latest doctrine branch updates in! Thanks, @skellamp!

**Pull request merged**
